File: app/src/main/java/com/example/gb_android_base_appnotes/data/CardsSource.java
package com.example.gb_android_base_appnotes.data;

public interface CardsSource {
    CardsSource init(CardsSourceResponse cardsSourceResponse);
    CardNote getNoteData(int position);
    int size();
    void deleteCardNote(int position);
    void updateCardNote(int position, CardNote cardNote);
    void addCardNote(CardNote cardNote);
    void clearCardNote();
}

File: app/src/main/java/com/example/gb_android_base_appnotes/data/CardsSource.kt
package com.example.gb_android_base_appnotes.data

interface CardsSource {
    fun init(cardsSourceResponse: CardsSourceResponse?): CardsSource?
    fun getNoteData(position: Int): CardNote?
    fun size(): Int
    fun deleteCardNote(position: Int)
    fun updateCardNote(position: Int, cardNote: CardNote?)
    fun addCardNote(cardNote: CardNote?)
    fun clearCardNote()
}